### DISCUSSION TITLE
Add possibility to show value of a Datetime Field in local time zone (Allow multi time zone usage of Data Studio)

### DISCUSSION THREAD

**Discussion by pbischoffcg** [directus/directus#19460] [Feature Requests]
### Summary

Every date time field should have a new "show in local time zone" option. If this option is deactivated the behavior is exactly like it is now. When this option is activated for a date time field, then the actual database value should be treated as UTC and for displaying the date time value Directus Data Studio it should be translate to the local time zone of the client (browser).

### Motivation

For every application, that has users in more than one time zone, a good pattern is to store datetime values in UTC in the database and let the client application (Browser, App,... whatever) translate the datetime value to its time zone. This pattern is currently not possible with the Directus Data Studio as client application.

### Detailed Design

As mentioned above the concept for datetime fields with enabled "show in local time zone" is to store all datetime values in UTC in the database. I think even the whole api server should treat this datetime values as UTC and send it as UTC datetime string in json responses. The client app should then take this UTC datetime and translate it to the local time zone of the browser, when displaying the value. It must be considered that this option not only concerns the displaying of values, but also the picking of datetimes. The picked datetime value has to be considered as in local time zone and has to be translated to UTC before sending it to the api server.
The possibility to enable or disable this new "show in local time zone" option should be placed in the "interface" and/or "display" menu item when creating a datetime field.

### Must Have:
- A possibility to enable/disable this new option per datetime field

### Drawbacks

I'm not sure if treating the actual datetime values of the database as UTC would be a new and maybe breaking behavior. If not then I don't know big drawbacks.

### Alternatives

Without this feature I see no good way to use the Data Studio in applications with users in more than one time zone.

### Adoption Strategy

If this new option is disabled by default then there should be nothing to do for adopting.

**Comment by cdwmhcc** [comment]
Great, thank you!

**Comment by bas-esbl** [comment]
I am not sure if I should do this by commenting, but wouldn't a 'timezone' setting be most logical? Eg setting your timezone on project level, or on user level? 

Alternatively somewhere in a config file. In previous versions of Directus, we were able to set the timezone in the project settings. Eg
`// Example: Setting the default time zone in Directus
const directus = new Directus('<your-instance-url>');
directus.settings.update({
  'project_timezone': 'UTC'
});`

Right now it is really hard to explain users they should always convert the current time to utc when for instance looking for records between 2 specific dates. This is very user unfriendly.

**Reply by FreeSynergy** [reply]
Every input or output date is coded to the local time. So you write the local time, it will be stored as utc. When you open the dataset, it will be coded again to your local time. 

Example: (Berlin +1)
Berlin time input: 15:00
UTC: 14:00
Somebody in Berlin opens Dataset: 15:00
I change the location to another timezone (-1): 14:00

The user should not convert the current time. Directus convert the inputtime to the utc.

**Comment by miklosdankhazi** [comment]
Any update on this?

**Comment by U11Leung** [comment]
Another yell for this implementation here, it is kind of necessity when using directus frontend  to navigate/edit data of datetime. Although frontend page is not always used to edit data, but trust me if one got to do that for a couple of times he will soon feel the need.

**Comment by JamesW1** [comment]
Thanks for the suggestion! We've moved feature requests to our public roadmap at https://roadmap.directus.io/roadmap. I've migrated this request there for you. If we see enough community demand, it will appear in our backlog.
